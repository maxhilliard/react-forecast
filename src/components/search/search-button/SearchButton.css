.searchButton {
    background-color: black;
    cursor: pointer;
}