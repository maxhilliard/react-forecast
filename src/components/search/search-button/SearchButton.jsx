import React from 'react';
import PropTypes from 'prop-types';

import styles from './SearchButton.css';

const SearchButton = ({ buttonText, handleSubmit, isDisabled }) => (
    <button disabled={isDisabled} className={styles.searchButton} type="button" onClick={handleSubmit}>
        {buttonText}
    </button>
);

SearchButton.propTypes = {
    buttonText: PropTypes.string,
    handleSubmit: PropTypes.func.isRequired,
    isDisabled: PropTypes.bool.isRequired,
};

SearchButton.defaultProps = {
    buttonText: 'Search',
};

export default SearchButton;
