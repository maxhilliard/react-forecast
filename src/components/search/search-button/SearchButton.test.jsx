import React from 'react';
import { shallow } from 'enzyme';

import SearchButton from './SearchButton';

describe('SearchButton', () => {
    const mockProps = {
        isDisabled: false,
        handleSubmit: () => {},
    };

    const wrapper = shallow(<SearchButton {...mockProps} />);

    describe('snapshot', () => {
        it('should match snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('props', () => {
        it('should render buttonText prop as expected', () => {
            const mockButtonText = 'mock-button-text';

            wrapper.setProps({
                ...mockProps,
                buttonText: mockButtonText,
            });

            expect(wrapper.find('button').text()).toEqual(mockButtonText);
        });
    });

    describe('event handling', () => {
        it('should invoke handleSubmit onClick', () => {
            const handleSubmitSpy = jest.fn();
            wrapper.setProps({ handleSubmit: handleSubmitSpy });

            const button = wrapper.find('button');
            button.simulate('click');

            expect(handleSubmitSpy).toHaveBeenCalled();
        });
    });
});
