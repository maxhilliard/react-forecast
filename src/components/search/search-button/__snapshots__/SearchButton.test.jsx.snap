// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchButton snapshot should match snapshot 1`] = `
<button
  className="searchButton"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Search
</button>
`;
